src/cpu/aarch64/jit_uni_eltwise.cpp

```diff
@@ -480,22 +484,23 @@ status_t jit_uni_eltwise_bwd_t<isa, d_type>::pd_t::init(engine_t *engine) {
     return ok ? status::success : status::unimplemented;
 }
 
-template <cpu_isa_t isa, data_type_t d_type>
-jit_uni_eltwise_bwd_t<isa, d_type>::jit_uni_eltwise_bwd_t(const pd_t *apd)
+template <cpu_isa_t isa>
+jit_uni_eltwise_bwd_t<isa>::jit_uni_eltwise_bwd_t(const pd_t *apd)
     : primitive_t(apd) {}
 
-template <cpu_isa_t isa, data_type_t d_type>
-jit_uni_eltwise_bwd_t<isa, d_type>::~jit_uni_eltwise_bwd_t() = default;
+template <cpu_isa_t isa>
+jit_uni_eltwise_bwd_t<isa>::~jit_uni_eltwise_bwd_t() = default;
 
-template <cpu_isa_t isa, data_type_t d_type>
-status_t jit_uni_eltwise_bwd_t<isa, d_type>::init(engine_t *engine) {
+template <cpu_isa_t isa>
+status_t jit_uni_eltwise_bwd_t<isa>::init(engine_t *engine) {
     CHECK(safe_ptr_assign(kernel_, new jit_uni_kernel_t<isa>(pd())));
     return kernel_->create_kernel();
 }
 
-template <cpu_isa_t isa, data_type_t d_type>
-status_t jit_uni_eltwise_bwd_t<isa, d_type>::execute(
+template <cpu_isa_t isa>
+status_t jit_uni_eltwise_bwd_t<isa>::execute(
         const exec_ctx_t &ctx) const {
+    const data_type_t data_t = pd()->use_dst() ? pd()->dst_md()->data_type : pd()->src_md()->data_type;
```

Comment:
This is a wrong use. This should be removed, pointers should be used as `char` or `void` with later cast to char to perform pointer arithmetic to shift them.

Reply:
I personally use `void` to make conscious decisions about further casting if/when it's needed, but it doesn't have to be like that. I'm fine with either type as long as this practice stretches to all implementations for the sake of consistency.